packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,295 @@
+        (task) => {
+          const {
+            name,
+            description,
+            lastCompleted,
+            dueDate,
+            interval,
+            assigned,
+            pool,
+            house,
+          } = task;
+
+          return {
+            name,
+            description,
+            lastCompleted,
+            dueDate,
+            interval,
+            assigned,
+            pool,
+            house,
+            isComplete: task.lastCompleted != null,
+          };
+        },
+      );
+      return {
+        tasks: updatedTasksDto,
+      };
+    }
+  }
+
+  @Delete('/:id')
+  @HttpCode(204)
+  @ApiParam({
+    name: 'id',
+    required: true,
+    description: 'id for the task to delete',
+  })
+  @ApiOperation({ summary: 'delete a task from a house.' })
+  @ApiNoContentResponse({
+    description: 'task successfully deleted.',
+  })
+  @ApiForbiddenResponse({
+    description: 'user is not the owner of the house',
+  })
+  @ApiNotFoundResponse({
+    description: 'task not found',
+  })
+  async deleteTaskFromHouse(@Param('id') id, @User() user: DecodedIdToken) {
+    if (!isValidObjectId(id)) {
+      throw new HttpException('task does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    const task = await this.taskStoreService.findOne(id);
+    if (!task) {
+      throw new HttpException('task not found', HttpStatus.NOT_FOUND);
+    }
+
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+    if (!userDoc.house) {
+      throw new HttpException(
+        'user is not in the house',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const house = await this.houseStoreService.findOne(userDoc.house);
+    if (!house.owner.equals(userDoc._id)) {
+      throw new HttpException(
+        'user is not the owner of the house',
+        HttpStatus.FORBIDDEN,
+      );
+    }
+
+    this.taskStoreService.delete(task._id);
+  }
+
+  @Put('/:id/completed')
+  @ApiParam({
+    name: 'id',
+    required: true,
+    description: 'id for the task to mark as complete or not',
+  })
+  @ApiOperation({
+    summary: 'mark user as having completed/not completed task.',
+  })
+  @ApiResponse({
+    description: 'tasks successfully marked as complete/not complete.',
+  })
+  @ApiBadRequestResponse({
+    description: 'user is not assigned to the task',
+  })
+  @ApiNotFoundResponse({
+    description: 'task not found',
+  })
+  async markTaskAsComplete(
+    @Body() completeTaskDto: CompleteTaskDto,
+    @Param('id') id,
+    @User() user: DecodedIdToken,
+  ) {
+    if (!isValidObjectId(id)) {
+      throw new HttpException('task does not exist', HttpStatus.NOT_FOUND);
+    }
+
+    const task = await this.taskStoreService.findOne(id);
+    if (!task) {
+      throw new HttpException('task not found', HttpStatus.NOT_FOUND);
+    }
+
+    if (task.assigned !== user.uid) {
+      throw new HttpException(
+        'user is not assigned to task',
```

Comment:
a little ambiguous, could mean this task or any task
`'user is not assigned to the task'` maybe?